packages/@jsii/kernel/src/kernel.ts

```diff
@@ -15,6 +15,28 @@ import * as onExit from './on-exit';
 import * as wire from './serialization';
 import * as tar from './tar-cache';
 
+export enum JsiiErrorType {
```

Comment:
We can use a `const enum` here -- this is a shared contract between the runtimes & the JavaScript kernel, and so in principle the string versions can no longer change without it being a breaking change across the board. At this stage, constant propagation (which is what `const enum` does in TypeScript) is what'll have the smallest performance impact.

```suggestion
export const enum JsiiErrorType {
```